src/kernel/boot.c

```diff
@@ -873,21 +881,64 @@ BOOT_CODE bool_t init_freemem(word_t n_available, const p_region_t *available,
      * from. If possible, retain any left over memory as an extra p region */
     word_t size = calculate_rootserver_size(it_v_reg, extra_bi_size_bits);
     word_t max = rootserver_max_size_bits(extra_bi_size_bits);
+    pptr_t start;
     for (; i >= 0; i--) {
-        word_t next = i + 1;
-        pptr_t start = ROUND_DOWN(ndks_boot.freemem[i].end - size, max);
-        if (start >= ndks_boot.freemem[i].start) {
-            create_rootserver_objects(start, it_v_reg, extra_bi_size_bits);
-            if (i < ARRAY_SIZE(ndks_boot.freemem)) {
-                ndks_boot.freemem[next].end = ndks_boot.freemem[i].end;
-                ndks_boot.freemem[next].start = start + size;
+        // SimplExportAnRefine has poor handling of array indices that are sums
```

Comment:
I recommend extending this comment to say this is about the proofs. For you putting `SimplExportAnRefine` into that context might be obvious, but for people with just a C view on the code this comment might feel a bit mysterious. Could say `"The proof's rule SimplExportAnRefine has poor ..."` if "rule" is the right word there.

Reply:
Yeah, good point. Although, it turned out this wasn't the problem, so my commit should be dropped anyway.